### PR TITLE
Adding timeout options both CLI and .cigar.json

### PR DESCRIPTION
**Pull request by WyriHaximus**
🎆🎆🎆 Happy New Year 🎆🎆🎆

This PR adds the timeout options discussed in #31
Closes / implements #31

Any values in the CLI are overridden by what ever value is in the `.cigar.json`

### CODE REVIEW

File: bin/cigar [hunk 1]
```diff
@@ -18,7 +18,7 @@ use Brunty\Cigar\Outputter;
 use Brunty\Cigar\Parser;
 use Brunty\Cigar\Result;
 
-$options = getopt('c:ia:u:jh:', ['version', 'help', 'quiet', 'config:', 'insecure', 'auth:', 'url:', 'json', 'header:']);
+$options = getopt('c:ia:u:jh:t:ct:', ['version', 'help', 'quiet', 'config:', 'insecure', 'auth:', 'url:', 'json', 'header:', 'timeout:', 'connect-timeout:']);
```
Comment (Brunty) [inline bin/cigar]: Shortopts are for single-character options only, so `ct` won't work (we also already have a `c` and a `t` option, so it'll produce results like this:

```
> php opts.php -c 'config_file.json' -t 'timeout' -ct '1'
array(2) {
  ["c"]=>
  array(2) {
    [0]=>
    string(16) "config_file.json"
    [1]=>
    string(1) "t"
  }
  ["t"]=>
  string(7) "timeout"
}
```

File: bin/cigar [hunk 2]
```diff
@@ -27,14 +27,16 @@ if (isset($options['help'])) {
...
-      \033[32m-h,           --header\033[0m             Custom header "\074name\076: \074value\076"
-      \033[32m-j,           --json\033[0m               Output JSON
-      \033[32m              --quiet\033[0m              Do not output any message
-      \033[32m              --version\033[0m            Print the version of Cigar 
+      \033[32m-c file.json, --config=file.json\033[0m         Use the specified config file instead of the default .cigar.json file 
+      \033[32m-u URL,       --url=URL\033[0m                  Base URL for checks, e.g. https://example.org/
+      \033[32m-i,           --insecure\033[0m                 Allow invalid SSL certificates
+      \033[32m-a,           --auth\033[0m                     Authorization header "\074type\076 \074credentials\076"
+      \033[32m-h,           --header\033[0m                   Custom header "\074name\076: \074value\076"
+      \033[32m-ct,          --connect-timeout=TIMEOUT\033[0m  Connect Timeout
```
Comment (Brunty) [inline bin/cigar]: Help output will need changing with the change of the `ct` option

File: spec/AsyncCheckerSpec.php [hunk 3]
```diff
@@ -62,4 +62,20 @@
             expect($results[0]->getStatusCode())->toEqual($expected[0]->getStatusCode());
         });
     });
+
```
Comment (Brunty) [inline spec/AsyncCheckerSpec.php]: I'd like a test adding for the command line options to ensure they're configured correcly I think

File: bin/cigar (hunk 1 above)
Comment (WyriHaximus) [reply bin/cigar]: Doh my bad, changed it :+1:

File: bin/cigar (hunk 2 above)
Comment (WyriHaximus) [reply bin/cigar]: Removed the mention of `ct` here

File: spec/AsyncCheckerSpec.php (hunk 3 above)
Comment (WyriHaximus) [reply spec/AsyncCheckerSpec.php]: Added the response timeout, not sure yet how to set up the connect timeout test. Might set up a small @reactphp project up that listens on a specific port to do something like that

Comment (Brunty) [reply spec/AsyncCheckerSpec.php]: I'm okay with the test being left for now, and an issue being filed to add it at a later time.

Comment (WyriHaximus) [reply spec/AsyncCheckerSpec.php]: Created a new issue for that as https://github.com/Brunty/cigar/issues/34

File: bin/cigar [hunk 4]
```diff
@@ -78,17 +80,19 @@ if ( ! file_exists($file)) {
     exit(1);
 }
 
+$secure = ! (isset($options['i']) || isset($options['insecure']));
+$authorization = $options['a'] ?? $options['auth'] ?? null;
+$headers = (array) ($options['h'] ?? $options['header'] ?? []);
+$connectTimeout = $options['connect-timeout'] ?? null;
```
Comment (Brunty) [inline bin/cigar]: This will always be `null` as `connect-timeout` is not in the options anymore (but it's still in the help text)

**Comment by WyriHaximus** [comment]
@Brunty pick a letter and I can use that one. But there isn't one that directly makes sense.

**Comment by WyriHaximus** [comment]
@Brunty ping, let me know if you have a preference for a letter I can use, otherwise I'll pick one using RNG later this week

**Comment by WyriHaximus** [comment]
@Brunty ok updated it with `s` as short for `connect-timeout`.

**Comment by WyriHaximus** [comment]
Assuming this would be 1.12, should I tag is later today?

**Comment by Brunty** [comment]
Please! I'll squash and merge now :)

**Comment by WyriHaximus** [comment]
Ok it's out: https://github.com/Brunty/cigar/releases/tag/1.12.0

**Pull request merged**